codec-http/src/main/java/io/netty/handler/codec/http/DefaultHttpHeaders.java

```diff
@@ -450,15 +243,39 @@ public long getTimeMillis(CharSequence name, long defaultValue) {
     }
 
     @Override
-    public List<Map.Entry<String, String>> entries() {
-        return headers.entriesConverted();
+    public List<Entry<String, String>> entries() {
+        if (isEmpty()) {
+            return Collections.emptyList();
+        }
+        List<Entry<String, String>> entriesConverted = new ArrayList<Entry<String, String>>(
+                headers.size());
+        for (Entry<String, String> entry : this) {
+            entriesConverted.add(entry);
+        }
+        return entriesConverted;
     }
 
+    /**
+     * @deprecated Use {@link #iteratorCharSequence()}.
+     */
     @Override
+    @Deprecated
     public Iterator<Map.Entry<String, String>> iterator() {
         return headers.iteratorConverted();
     }
 
+    /**
+     * @deprecated Future major releases will have this be the default iterator.
```

Comment:
I don't have a preference but I thought I remember @trustin leaving a comment that he likes them at the top. @trustin ?